### PR TITLE
roachtest: try more nodes for 'debug zip' if the first fails

### PR DESCRIPTION
**Pull request by knz**
Release note: None

### CODE REVIEW

**Review by tbg — APPROVED** [review]
Thanks @knz! Were you able to take this for a test ride?

File: pkg/cmd/roachtest/cluster.go [hunk 1]
```diff
@@ -1376,14 +1376,23 @@ func (c *cluster) FetchDebugZip(ctx context.Context) error {
...
+			output, err := execCmdWithBuffer(ctx, c.l, roachprod, "ssh", c.name+":"+si, "--",
+				"./cockroach", "debug", "zip", "--url", "{pgurl:"+si+"}", zipName)
+			if err != nil {
+				c.l.Printf("./cockroach debug zip failed: %s", output)
+				if i < c.spec.NodeCount {
+					continue
+				}
+				return err
+			}
+			return execCmd(ctx, c.l, roachprod, "get", c.name+":1", zipName /* src */, path /* dest */)
```
Comment (tbg) [inline pkg/cmd/roachtest/cluster.go:1394]: needs to use `si` here instead of `1`

**Comment by knz** [comment]
I was hoping CI to do the ride for me... but now I'll test it too

**Comment by tbg** [comment]
CI will ride the happy case, but not the one in which it actually does
something unfortunately

**Comment by knz** [comment]
ok I tested it, and it does the job.
Thanks again!

bors r=tbg

**Pull request merged**